pkg/apiserver/negotiate_test.go

```diff
@@ -41,27 +41,34 @@ func (n *fakeNegotiater) SupportedStreamingMediaTypes() []string {
 	return n.streamTypes
 }
 
-func (n *fakeNegotiater) SerializerForMediaType(mediaType string, options map[string]string) (runtime.Serializer, bool) {
+func (n *fakeNegotiater) SerializerForMediaType(mediaType string, options map[string]string) (runtime.SerializerInfo, bool) {
 	n.mediaType = mediaType
 	if len(options) > 0 {
 		n.options = options
 	}
-	return n.serializer, n.serializer != nil
+	return runtime.SerializerInfo{Serializer: n.serializer, MediaType: n.mediaType, EncodesAsText: true}, n.serializer != nil
 }
 
-func (n *fakeNegotiater) StreamingSerializerForMediaType(mediaType string, options map[string]string) (runtime.Serializer, runtime.Framer, string, bool) {
+func (n *fakeNegotiater) StreamingSerializerForMediaType(mediaType string, options map[string]string) (runtime.StreamSerializerInfo, bool) {
 	n.streamMediaType = mediaType
 	if len(options) > 0 {
 		n.streamOptions = options
 	}
-	return n.streamSerializer, n.framer, mediaType, n.streamSerializer != nil
+	return runtime.StreamSerializerInfo{
+		SerializerInfo: runtime.SerializerInfo{
+			Serializer:    n.serializer,
+			MediaType:     mediaType,
+			EncodesAsText: true,
+		},
+		Framer: n.framer,
```

Comment:
None of the tests are currently using it - can be added, but it wouldn't do anything